omnipaxos/src/sequence_paxos/leader.rs

```diff
@@ -343,19 +334,32 @@ where
                     .set_decided_idx(decided_idx)
                     .expect(WRITE_ERROR_MSG);
                 for pid in self.leader_state.get_promised_followers() {
-                    match self.leader_state.get_batch_accept_meta(pid) {
-                        Some((bal, msg_idx)) if bal == self.leader_state.n_leader => {
-                            let PaxosMessage { msg, .. } = self.outgoing.get_mut(msg_idx).unwrap();
-                            match msg {
-                                PaxosMsg::AcceptDecide(acc) => acc.decided_idx = decided_idx,
-                                _ => panic!("Cached index is not an AcceptDecide!"),
-                            }
-                        }
-                        _ => self.send_decide(pid, decided_idx, false),
-                    };
+                    let buffered_accdec = self.get_buffered_accdec_message(pid);
+                    match buffered_accdec {
+                        Some(accdec) => accdec.decided_idx = decided_idx,
+                        None => self.send_decide(pid, decided_idx, false),
+                    }
+                }
+            }
+        }
+    }
+
+    fn get_buffered_accdec_message(&mut self, to: NodeId) -> Option<&mut AcceptDecide<T>> {
+        if let Some((bal, outgoing_idx)) = self.leader_state.get_batch_accept_meta(to) {
```

Comment:
we should update the function in leader state to also be called `latest`
```suggestion
        if let Some((bal, outgoing_idx)) = self.leader_state.get_latest_accept_meta(to) {
```